release.sh

```diff
@@ -2,32 +2,63 @@
 
 set -e
 
-last_tag=$(git tag --sort=-creatordate | head -n 1)
+last_tag=$(git tag | head -n 1)
 new_tag=$(semver bump patch "$last_tag")
 timestamp=$(date -u +%Y%m%d%H%M%S)
 ref=$(echo "$BRANCH" | sed -e "s/[^a-zA-Z0-9]//g")
 new_version=$(if [ "$ref" = "master" ]; then echo "${new_tag}"; else echo "${new_tag}-${ref}.$timestamp"; fi)
 
-echo "Bump version: ${last_tag} -> ${new_version}"
-git tag "${new_version}"
+validateFileChanges() {
+  #Define sensible file
+  sensible_files=(Dockerfile .dockerignore requirements.txt)
 
-# Create image tag
-image_tag="${IMAGE_NAME}:${new_version}"
-docker tag "${IMAGE_NAME}" "${image_tag}"
+  #Validate if sensible file are in list
+  for Item in ${sensible_files[*]} ;
+  do
+    compare_items=$(git diff --name-only HEAD~1..HEAD | grep $Item | grep -v example)
+    if [ "$Item" = "$compare_items" ] ; then
+        validateBranch
```

Comment:
There still isn't any guarantee right now that this line is executed just once - always. And also it's less intuitive just looking at the code. 

I would propose something to make it more understandable:
Instead of calling `validateBranch` from inside the function - can you make the `validateFileChange()` function to return a boolean results `0` or `1`. That is it returns a `1` value only if the changes are validated. 

And in the line 62 where you invoke it you may do something like this:
```bash
is_change_valid = validateFileChanges

if [ "$is_change_valid" = "1" ]; then
    validateBranch
fi
```

Reply:
The proposed option also does not ensure that it is executed only once. In the code if you see in line `33` and `57`, there are an `exit 0` indicating that the execution has finished. 
The `exit 0` is placed in the final step of both cases of execution.
The `exit 0` of the line `57` avoids that once the image in the first line coincides with the loop, it leaves the loop and ends the execution.

Reply:
Sorry, perhaps I didn't understand it properly. But what I'm proposing is that you extract the logic of validating the changes in one function `validateFileChanges` so that it only does one thing - check and return boolean.

Could you make that change and and then you can use just the function to do a one **if** check outside like I've shown you above in the example - you might not even need `validateBranch` function if you just extract it's code under the if condition. 

Please make that change and we can test it.